praw/objector.py

```diff
@@ -24,6 +46,54 @@ def kind(self, instance):
             if isinstance(instance, self.parsers[key]):
                 return key
 
+    def _objectify_dict(self, data):
+        """Create RedditBase objects from dicts.
+
+        :param data: The structured data, assumed to be a dict.
+        :returns: An instance of :class:`~.RedditBase`.
+
+        """
+        if ({'date', 'id', 'name'}.issubset(data)
+           or {'id', 'name', 'permissions'}.issubset(data)):
+            parser = self.parsers[self._reddit.config.kinds['redditor']]
+            return parser.parse(data, self._reddit)
+        if ({'conversation', 'messages', 'user', 'modActions'}.issubset(data)):
+            parser = self.parsers['ModmailConversation']
+            return parser.parse(data, self._reddit)
+        if {'actionTypeId', 'author', 'date'}.issubset(data):
+            # Modmail mod action
+            data = self._snake_case_keys(data)
+            parser = self.parsers['ModmailAction']
+            return parser.parse(data, self._reddit)
+        if {'bodyMarkdown', 'isInternal'}.issubset(data):
+            # Modmail message
+            data = self._snake_case_keys(data)
+            parser = self.parsers['ModmailMessage']
+            return parser.parse(data, self._reddit)
+        if {'isAdmin', 'isDeleted'}.issubset(data):
+            # Modmail author
+            data = self._snake_case_keys(data)
+            # Prevent clobbering base-36 id
+            del data['id']
+            data['is_subreddit_mod'] = data.pop('is_mod')
+            parser = self.parsers[self._reddit.config.kinds['redditor']]
+            return parser.parse(data, self._reddit)
+        if {'banStatus', 'muteStatus', 'recentComments'}.issubset(data):
+            # Modmail user
+            data = self._snake_case_keys(data)
+            data['created_string'] = data.pop('created')
+            parser = self.parsers[self._reddit.config.kinds['redditor']]
+            return parser.parse(data, self._reddit)
+        if {'displayName', 'id', 'type'}.issubset(data):
+            # Modmail subreddit
+            data = self._snake_case_keys(data)
+            parser = self.parsers[self._reddit.config.kinds[data['type']]]
+            return parser.parse(data, self._reddit)
+        if 'user' in data:
```

Comment:
If all of the above are `elif` I think this if and the subsequent `return data` can be inside an `else` scope.